en/docs/devops-and-ci-cd/configure-vpns-on-the-choreo-cloud-data-plane.md

```diff
@@ -0,0 +1,210 @@
+
+![Deployment architecture](../assets/img/devops-and-ci-cd/tailscale/deployment-architecture.png)
+
+Let's take a look at the specifics of each part to understand the deployment architecture.
+
+**Choreo project**
+
+In Choreo, a project groups various components. For more information on what a project in Choreo is, see the documentation on [Project](../choreo-concepts/project.md).
+
+**Tailscale proxy**
+
+This acts as the Tailscale pre-installed forward proxy, facilitating secure peer-to-peer WireGuard connections from the Choreo cloud data plane to private networks. It includes a [Tailscale Daemon](https://tailscale.com/kb/1278/tailscaled), [SOCKS5 proxy](https://tailscale.com/kb/1112/userspace-networking#socks5-vs-http), and a configurable TCP forwarder.
+
+**Tailscale daemon**
+
+This is the core component of Tailscale. It is a software service that provides secure network connectivity and private networking solutions. For more details see the [Tailscale documentation](https://tailscale.com/kb/1278/tailscaled).
+
+**SOCKS5 proxy**
+
+This uses Tailscale’s [userspace networking](https://tailscale.com/kb/1112/userspace-networking) mode, rather than the kernel mode. Therefore, the inbuilt SOCKS5 proxy handles the forwarded traffic and directs it through the Tailscale network.
+
+**TCP forwarder**
+
+Forwards inbound TCP (transmission control protocol) traffic from the Tailscale proxy container’s network interface to the SOCKS5 proxy, ensuring it reaches its destination via the secured WireGuard tunnel.
+
+**User applications and the Choreo API gateway**
+
+User applications within the same namespace (project) can use the Kubernetes service created to front the Tailscale proxy, for connecting to the corresponding private endpoints. You can either expose this service within the organization via the internal API gateway or expose it to the public via the external API gateway. For more details, see the documentation on [Choreo endpoints](../develop-components/configure-endpoints.md).
+
+<hr>
+
+Now that you understand the deployment architecture, let’s explore how you can use Tailscale to secure connections to your private networks.
+
+## Configure and use Tailscale to access private network endpoints
+
+This section walks you through the steps to create, configure, deploy, and use the Tailscale proxy component. 
+  
+![Tailscale proxy deployment](../assets/img/devops-and-ci-cd/tailscale/tailscale-proxy-deployment.png)
+
+Let's get started.
+
+### Prerequisites
+
+- Understand the basics of [how Tailscale works](https://tailscale.com/blog/how-tailscale-works).
+- Have a Tailscale account (Tailnet). There are multiple plans available for you to set up your Tailscale network. For details, see [Tailscale plans](https://tailscale.com/pricing).
+- Install Tailscale and connect your private services and database server to it. To quickly get started with Tailscale, see the [Tailscale quickstart](https://tailscale.com/kb/1017/install).
```

Comment:
Applied suggestion with a minor change because the comma there doesn't seem correct.
 Install Tailscale and connect your private data center, server to it, - - > Install Tailscale and connect your private data center or server to it,